### PR TITLE
add discord notification block

### LINKED ISSUE #8727: Ability to add a Discord webhook url while creating a new notification
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

Currently there are options for Mattermost, Microsoft Teams, Opsgenei, Pager Duty and Slack webhook to receive notifications. 

### Describe the proposed behavior

User will be able to select "Discord webhook" and paste the webhook url for it to create notification. The user will also be able to select "Run states" and "Notify Type (Optional)".

### Additional context

I am working on this feature and any suggestions / feedback / ideas are welcomed.

### PR DESCRIPTION
**Pull request by vishalsanfran**
Screenshot 2023-08-12 at 11 04 21 PM

Adds a DiscordWebhook notification block using the AbstractAppriseNotificationBlock

Closes #8727

### Example
```
In [1]: from prefect.blocks.notifications import DiscordWebhook

In [2]: discord_webhook_block = DiscordWebhook.load("discord-prefect")

In [3]: discord_webhook_block.notify("Hello from Prefect!!")
```

Screenshot 2023-08-12 at 11 11 50 PM

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [x] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

**Comment by vishalsanfran** [comment]
@zzstoatzz thanks, updated PR to change mattermost references to discord

**Comment by vishalsanfran** [comment]
@zzstoatzz I had to use a different logo image as that the one provided was not square in shape and that was failing the tests.
Please allow the other workflows to run.

File: src/prefect/blocks/notifications.py [hunk 1]
```diff
@@ -490,6 +490,103 @@ def block_initialization(self) -> None:
...
+        description=(
+            "Include an image in-line with the message describing the notification"
+            " type."
+        ),
+    )
+
+    avatar: bool = Field(
+        default=False,
+        description=(
+            "Over-ride the default discord avatar icon and replace it with one identify"
```
Comment (zzstoatzz) [inline src/prefect/blocks/notifications.py:558]:
```suggestion
            "Whether to override the default discord avatar icon."
```

File: src/prefect/blocks/notifications.py [hunk 2]
```diff
@@ -490,6 +490,103 @@ def block_initialization(self) -> None:
...
+
+    tts: bool = Field(
+        default=False,
+        description="Enable Text-To-Speech.",
+    )
+
+    include_image: bool = Field(
+        default=False,
+        description=(
+            "Include an image in-line with the message describing the notification"
```
Comment (zzstoatzz) [inline src/prefect/blocks/notifications.py:550]:
```suggestion
            "Whether to include an image in-line with the message describing the notification"
```

File: src/prefect/blocks/notifications.py [hunk 3]
```diff
@@ -490,6 +490,103 @@ def block_initialization(self) -> None:
...
+        description=(
+            "Identify the name of the bot that should issue the message. If one isn't"
+            " specified then the default is to just use your account (associated with"
+            " the incoming-webhook)."
+        ),
+    )
+
+    tts: bool = Field(
+        default=False,
+        description="Enable Text-To-Speech.",
```
Comment (zzstoatzz) [inline src/prefect/blocks/notifications.py:544]:
```suggestion
        description="Whether to enable Text-To-Speech.",
```

**Review by zzstoatzz — CHANGES_REQUESTED** [review]
nice! looking good, just a couple nits on fields descriptions

File: src/prefect/blocks/notifications.py (hunk 3 above)
Comment (vishalsanfran) [reply src/prefect/blocks/notifications.py]: updated

File: src/prefect/blocks/notifications.py (hunk 2 above)
Comment (vishalsanfran) [reply src/prefect/blocks/notifications.py]: updated

**Comment by zzstoatzz** [comment]
@vishalsanfran cool. We're [failing a pre-commit check](https://github.com/PrefectHQ/prefect/actions/runs/5932236782/job/16085676702?pr=10394) (do `pre-commit run .` locally) but once that's passing I think this is good to go!

**Comment by vishalsanfran** [comment]
Sorry, this got closed accidentally as I was trying to rebase

**Review by zzstoatzz — APPROVED** [review]
nice! thank you for contributing

ruff is complaining about an unused import unrelated to this PR's changes, similarly the failing test is not related to this PR

**Comment by vishalsanfran** [comment]
thanks! I'll look for more issues, but if there is anything specific you need help with, feel free to tag me.

**Pull request merged**